### PR TITLE
Fix hardcoded group names in VaccinationAsyncTask shown on the Register page for upcoming statuses

### PR DESCRIPTION
**Pull request by ekigamba**
This removes group name dependence on `VaccinationUtils.stateKey(String)` which uses hardcoded switch case for the vaccine groupings to now use preloaded group-names from the ImmunizationLibrary as per https://github.com/OpenSRP/opensrp-client-immunization/commit/37668d9dc1ba654ce7f3f00ef3e256575b6affcc

### CODE REVIEW

File: opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java [hunk 1]
```diff
@@ -514,8 +529,10 @@ private void updateRecordVaccination(VaccineViewRecordUpdateWrapper updateWrappe
 
     private String localizeStateKey(String stateKey) {
```
Comment (ndegwamartin) [inline opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: We can now do reverse look ups for string translations for this function

Comment (ekigamba) [reply opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: Huuh! What do you mean by this?

Comment (ndegwamartin) [reply opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: Basically use the method [here](https://developer.android.com/reference/android/content/res/Resources.html#getIdentifier(java.lang.String,%20java.lang.String,%20java.lang.String)) at runtime to retrieve the actual resource integer id for the key provided

Comment (ekigamba) [reply opensrp-child/src/main/java/org/smartregister/child/task/VaccinationAsyncTask.java]: Cool, I implemented that

**Pull request merged**